mailer/engine.py

```diff
@@ -112,6 +113,25 @@ def release_lock(lock):
     logging.debug("released.")
 
 
+def handle_backend_exception(connection, message, err):
+    """ simple error handler that marks messages failing due to SMTP
+    issues as deferred.  Only need to return the action type """
+    error_type = type(err)
+    action = None
```

Comment:
`action` needs to be `'deferred'` or `'sent'` to work with the rest of this pull request.